api/secret.go

```diff
@@ -113,69 +146,19 @@ func (s *Secret) TokenPolicies() ([]string, error) {
 		return s.Auth.Policies, nil
 	}
 
-	if s.Data == nil || s.Data["policies"] == nil {
+	if s.Data == nil {
 		return nil, nil
 	}
 
-	var tokenPolicies []string
-
-	// Token policies
-	{
-		_, ok := s.Data["policies"]
-		if !ok {
-			goto TOKEN_DONE
-		}
-
-		sList, ok := s.Data["policies"].([]string)
-		if ok {
-			tokenPolicies = sList
-			goto TOKEN_DONE
-		}
-
-		list, ok := s.Data["policies"].([]interface{})
-		if !ok {
-			return nil, fmt.Errorf("unable to convert token policies to expected format")
-		}
-		for _, v := range list {
-			p, ok := v.(string)
-			if !ok {
-				return nil, fmt.Errorf("unable to convert policy %v to string", v)
-			}
-			tokenPolicies = append(tokenPolicies, p)
-		}
+	tokenPolicies, err := s.extractListOfStrings("policies")
+	if tokenPolicies == nil || err != nil {
+		return tokenPolicies, err
```

Comment:
There are cases in the current code where we goto TOKEN_DONE despite having an empty tokenPolicies.  I think this `if`'s condition should be just `err != nil`.

Reply:
Yeah, I didn't notice it. Now the latest version of the code should be equivalent in functionality